ext/com_dotnet/com_variant.c

```diff
@@ -82,7 +82,7 @@ static void safe_array_from_zval(VARIANT *v, zval *z, int codepage)
 	return;
 
 bogus:
-	php_error_docref(NULL, E_WARNING, "COM: converting from PHP array to VARIANT array; only arrays with numeric keys are allowed");
+	zend_value_error("COM: Can only convert PHP array to VARIANT array if all keys are numeric");
```

Comment:
I'm not sure about this warning promotion. Is there a simple and efficient way to check in advance that the array has no string keys?

Reply:
Not really, except using `array_value()`

Reply:
So maybe stick with the warning for now?